### PR TITLE
Decoupling browser logic from token and url logic (no functional changes)

### PR DESCRIPTION
**Pull request by diegotsutsumi**
I'm breaking apart the badly written `client.py` into other classes, so we can move forward with the implementation of the BankAuthService. This is purely refactor, there is no functional changes in the application. This will enable us to more easily implement the different endpoints for the Bank Auth Service [described here](https://github.com/in-fin-neat/in-fin-neat-core/blob/master/docs/authorization-sequence.png)

Main changes:

1. There is a new interface called `BankClient` and an implementation for our gocardless/nordigen client called `GocardlessClient`
2. Coordination of Store and Token is now handled by `BankClientStoreTokenHandler`
3. Coordination of fetching URL/Requisitions is now done by `GocardlessUrlClientStoreHandler`. Requisition is very specific to gocardless I'm leaving some TODOs for a next PR to refactor it.

I'm still lacking tests, posting this PR as a draft to get early reviews.

### CODE REVIEW

File: personal_finances/bank_interface/client.py [hunk 1]
```diff
@@ -1,221 +0,0 @@
-from __future__ import annotations
```
Comment (diegotsutsumi) [inline personal_finances/bank_interface/client.py:1]: This is the file broken into multiple others

File: tests/bank_interface/test_bank_client.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            "transactions": {
+                "booked": ["transaction_b31"],
+                "pending": ["transaction_p31"],
+            }
+        },
+        {
+            "transactions": {
+                "booked": ["transaction_b41", "transaction_b42"],
+                "pending": ["transaction_p41"],
```
Comment (leonardohscastilho) [inline tests/bank_interface/test_bank_client.py:43]: What about testing empty lists for "booked" and "pending". I know its gonna be difficult to happen a user has no booked transaction, but for pending may can happen

**Review by leonardohscastilho — COMMENTED** [review]
Besides the two comments that I would like to know your thoughts on, the rest looks great to me.

File: personal_finances/bank_interface/bank_client.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        )
+
+    def generate_new_token(self) -> TokenObject:
+        new_gocardless_token = log_wrapper(self._nordigen_client.generate_token)
+        return gocardless_token_adapter(new_gocardless_token)
+
+    def get_authorization_url(
+        self, bank_details: BankDetails, validation_provider: BankValidationProvider
+    ) -> AuthorizationUrl:
+        institution_id = log_wrapper(
```
[the remark this answers is not in the document]
Comment (diegotsutsumi) [reply personal_finances/bank_interface/bank_client.py:96]: Your suggestion is great, I was more focused on the split and bigger redactors didn't want to scope creep this PR. If you agree we can do it later, we can keep the ball rolling for now.

There's a small caveat that needs better thinking on the decorator solution, you wouldn't be able to choose when to log and when not to log. Always logging is ok, but we need to at least differentiate who is the caller in the log so we keep the same troubleshooting level as today.

File: tests/bank_interface/test_bank_client.py (hunk 2 above)
Comment (diegotsutsumi) [reply tests/bank_interface/test_bank_client.py:43]: Good one, I'll add it in a next revision

**Review by leonardohscastilho — APPROVED** [review]

**Pull request merged**